### ISSUE TITLE
[Question] Trying to understand permissions in order to create UI for managing ACL

### ISSUE THREAD

**Issue by jameswagoner** [yajra/laravel-acl#25]
Managing Roles are straightforward; `name`, `slug`, and a `description`. Though it's not clear what `system` is for.

Permissions is where I feel is not fully documented. Don't see any information on this except in a Test Case. Also while reading through source code, I see `HasPermission` trait. Why is that not added to the `User` model for retrieving permissions of a user or is this optional and one would only know if they happened to dig through source code?

Two actual questions; What mechanism is there for creating a permission other than using the create facade on Permission model? What is `resource` and `system` referring two and how is it being used during authorization?

**Comment by yajra** [comment]
A quick answer would be:

`system` - Is a flag I used to protect the record from being deleted in the UI. I usually set this to true if the permission was added via seeder. Any new permission added via UI is set to false.

`resource` - is a permission group that are used for route resource authorization. See this https://github.com/yajra/cms-core/blob/4.0/src/Http/Controllers/RolesController.php#L26 for actual implementation. Having a resource permission and using it on the controller will protect all your routes using a single line of code.

I do agree that documentation is not yet refined and need some more enhancements but I haven't got the chance to update it yet.
